File: Leetcode/isPalindrome.java
public boolean isPalindrome(String s) {
        if (s == null || s.trim().length() == 0) {
            return true;
        }
        String str = s.trim().toLowerCase();
        int i =0; 
        int j = str.length() - 1;
        while(i <= j) {
            char c = str.charAt(i);
            char ch = str.charAt(j);
            if (!Character.isLetterOrDigit(c)) {
                i++;
            } else if (!Character.isLetterOrDigit(ch)) {
                j--;
            } else if (ch == c) {
                i++;
                j--;
            } else {
                return false;
            }
        }
        return true;
    }

File: Leetcode/isPalindrome.swift
func isPalindrome(s: String) -> Bool {
    let lowerStr = s.lowercaseString
    let len = s.characters.count
    var i = 0
    var j = len - 1
    while i < j {
        //omit the non alphanumric character
        var ch = lowerStr[lowerStr.startIndex.advancedBy(i)]
        var c =  lowerStr[lowerStr.startIndex.advancedBy(j)]
        if !(ch >= "a" && ch <= "z" || ch >= "0" && ch <= "9") {
            i++
        } else if !(c >= "a" && c <= "z" || c >= "0" && c <= "9") {
            j--
        } else if ch == c {
            i++
            j--
        } else {
            return false
        }
    }
    return true
}
